emily/handler/tests/integration/complex/reorg.rs

```diff
@@ -0,0 +1,157 @@
+    // Extract the scenario.
+    scenario.validate().unwrap();
+    let ReorgScenario {
+        initial_chain_length,
+        reorg_depth: _,
+    } = scenario;
+
+    // Setup the test client and environment.
+    let client: TestClient = TestClient::new();
+    client.setup_test().await;
+
+    // Identifier for the current fork.
+    let mut fork_id: u32 = 0;
+
+    // Step 1: Make an initial fork.
+    // -------------------------------------------------------------------------
+
+    // Process some deposits and withdrawals.
+    for stacks_block_height in 0..initial_chain_length {
+        // Setup requests.
+        let chainstate = util::test_chainstate(stacks_block_height, fork_id);
+        let stacks_block_hash = chainstate.stacks_block_hash.clone();
+        let deposit_request = CreateDepositRequestBody {
+            bitcoin_txid: format!("{TEST_BITCOIN_TXID}-{stacks_block_height}"),
+            bitcoin_tx_output_index: 1,
+            reclaim: TEST_RECLAIM_SCRIPT.to_string(),
+            deposit: TEST_DEPOSIT_SCRIPT.to_string(),
+        };
+        let withdrawal_request = CreateWithdrawalRequestBody {
+            request_id: stacks_block_height,
+            stacks_block_hash: stacks_block_hash,
+            recipient: TEST_RECIPIENT.to_string(),
+            amount: 1,
+            parameters: TEST_WITHDRAWAL_PARAMETERS.clone(),
+        };
+
+        // Make requests to populate the database.
+        client.create_chainstate(&chainstate).await;
+        client.create_deposit(&deposit_request).await;
+        client.create_withdrawal(&withdrawal_request).await;
+    }
+
+    // Ensure that all right number of deposits and withdrawals were created.
+    let all_deposits = client.get_all_deposits().await;
+    let all_withdrawals = client.get_all_withdrawals().await;
+
+    // Ensure that the right number of deposits and withdrawals were made.
+    assert_eq!(all_deposits.len(), initial_chain_length as usize);
+    assert_eq!(all_withdrawals.len(), initial_chain_length as usize);
+
+    // Verify that the chain tip is the highest block created.
+    let chain_tip: Chainstate = client.get_chaintip().await;
+    assert_eq!(
+        chain_tip,
+        util::test_chainstate(initial_chain_length - 1, fork_id),
+    );
+
+    // // Step 2: Create a conflicting fork.
+    // // -------------------------------------------------------------------------
+    // fork_id += 1;
+
+    // // Set a conflicting chainstate for a lower than top depth to initiate an internal reorg.
+    // let reorganized_chainstates = scenario.reorganized_chainstates(fork_id);
+    // let lowest_reorganized_block: Chainstate = reorganized_chainstates.first().unwrap().clone();
+    // client.update_chainstate(&lowest_reorganized_block).await;
+
+    // // Verify that the chain tip is updated to be the new reorg height.
+    // let chain_tip: Chainstate = client.get_chaintip().await;
+    // assert_eq!(chain_tip, lowest_reorganized_block);
```

Comment:
Is all of this supposed to be commented out?

Reply:
Yep, I'm going to comment it back in since this logic isn't implemented yet.

Reply:
Since this is for a future PR do you think I should delete it or keep it?

Reply:
Keep it, I was just wondering if it was an oversight or something.